src/components/ColorPicker/CustomColorPicker.tsx

```diff
@@ -32,21 +30,25 @@ export const CustomColorPicker: FC<Omit<ColorPickerProps, "palette">> = ({
     onSelect,
 }) => {
     const colorFormats = Object.values(ColorFormat).map((id) => ({ id, title: id.toLocaleUpperCase() }));
-    const [{ hsl, hsv, rgb, hex }, setColor] = useState(transformColor(currentColor));
+    const [color, setColor] = useState<Color>(currentColor);
+    const parsedColor = tinycolor(color);
+    const { r, g, b, a } = parsedColor.toRgb();
 
     useEffect(() => {
-        setColor(transformColor(currentColor));
+        setColor(currentColor);
     }, [currentColor]);
 
-    const [hexInput, setHexInput] = useState(hex.substring(1));
+    const [hexInput, setHexInput] = useState(parsedColor.toHex());
 
     useEffect(() => {
-        setHexInput(hex.substring(1));
-    }, [hex]);
+        setHexInput(parsedColor.toHex());
```

Comment:
You could move this inside the `useEffect` from above since if the `currentColor` changes, so does `color`

Reply:
Changed